target/linux/mediatek/image/mt7622.mk

```diff
@@ -50,7 +52,7 @@ define Build/m32-recovery-header-kernel1
 	echo -en "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" >> "$@.header"
 # Note: The last 2 bytes of the following line are the checksum of the header
 # If any data in the header will be changed, the checksum must be re-calculated
```

Comment:
is this comment still needed?

Reply:
Doesn't make sense it the way it currently is. Tried to make something more useful out of it.

Reply:
@RolandoMagico 
I thought it makes sense to compute the checksum in the makefile, but something doesn't work for me:
```sh
header='DLK6E6010001\x00\x00\xCF\x33\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x8D\x57\x30\x0B\x00\x00\x2C\x00\x00\x00\xD0\x02\x00\x00\x2C\x00\x00\x00\xD0\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x42\x48\x02\x00\x00\x00\x08\x00\x00\x00\x00\x00\x60\x6E\x68\x61'
echo -ne $header | od -v -w2 -tu2 -An --endian little | paste -sd+ | bc | xargs printf '%04x'
```
prints `2fff4`, not `0000`. Same result for the other model.
I tried big endian instead, still no go.

Reply:
So it's sum mod 65535. not mod 65536. How about this:
```sh
# create $@.header without the checksum
od -v -w2 -tu2 -An --endian little $@.header | awk '{s+=$1;}END{s=65535-s%65535;printf "%c%c",s%256,s/256;}' >> $@.header
```

Reply:
I tried it:
```
# Only zeros
	echo -en "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00" >> "$@.header"
# Last 16 bytes, but without checksum
	echo -en "\x42\x48\x02\x00\x00\x00\x08\x00\x00\x00\x00\x00\x60\x6E" >> "$@.header"
# Calculate and append checksum
	od -v -w2 -tu2 -An --endian little $@.header | awk '{s+=65535-$1;}END{s%=65535;printf "%c%c",s%256,s/256;}' > $@.header.checksum
	cat "$@.header" "$@.header.checksum" "$@" > "$@.new"
	mv "$@.new" "$@"
	rm "$@.header" "$@.header.checksum"
```
But checksum is 00 00 for both devices.

Reply:
You don't want `s = 65535 - s` since checksum 0 needs to remain 0. It can be `s = (65535 - s) % 65535` though.
And if you insist on doing the modulo every time, `s+=$1; s%=65535;` makes more sense.

How about
```sh
# Calculate and append checksum
( \
	cat "$@.header"; \
	od -v -w2 -tu2 -An --endian little $@.header | awk '{s+=65535-$1;}END{s%=65535;printf "%c%c",s%256,s/256;}'; \
	cat "$@"; \
) > "$@.new"
```

Reply:
> I want to simulate 16bit overflows using the `s%=65536`

If you want it every numberyou could do either
```sh
awk '{s+=$1;s%=65535;}END{s=65535-s;printf "%c%c",s%256,s/256;}'
```
or
```sh
awk '{s+=65535-$1;s%=65535;}END{printf "%c%c",s%256,s/256;}'
```
My version does it once, at the end.

Reply:
Ok, I think now I understood your script. All three versions from you are working on command line printing "ha" (0x68 0x61) for the M32, but if I'm using it in the Makefile, I get only 00 00 in the binary. I also see that the file $@.header.checksum already contains the incorrect values.

Reply:
Escaping $ in $1 to $$1 (Makefiles...) was missing